buildSrc/src/main/groovy/org/gradle/build/UpdateReleasedVersions.groovy

```diff
@@ -21,24 +21,43 @@ import com.google.common.base.Preconditions
 import groovy.json.JsonBuilder
 import groovy.json.JsonSlurper
 import org.gradle.api.DefaultTask
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.TaskAction
 import org.gradle.util.GradleVersion
 
+import javax.inject.Inject
+
 class UpdateReleasedVersions extends DefaultTask {
 
     @Internal
     File releasedVersionsFile
 
     @Internal
-    ReleasedVersion currentReleasedVersion
+    Property<ReleasedVersion> currentReleasedVersion
+
+    @Inject
+    UpdateReleasedVersions(ObjectFactory objectFactory) {
+        currentReleasedVersion = objectFactory.property(ReleasedVersion)
+    }
+
+    void setReleasedVersion(Provider<ReleasedVersion> provider) {
```

Comment:
I thought we were only going to expose `getReleasedVersion()`, so you can call `getReleasedVersion().set...`. IIUC, the setter would then be generated at runtime.

See `ValidateTaskProperties.outputFile` https://github.com/gradle/gradle/blob/4a6d67f76300480fa02b3b5d87df6bee371dcf10/subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/ValidateTaskProperties.java#L314-L323 and its usage in the Gradle build:https://github.com/gradle/gradle/blob/0de158793b915323a965c90a35a0eec319ff1eed/gradle/taskProperties.gradle#L6-L8.

Reply:
done. Kinda forgot about it.